src/hu_nmt/data_augmentator/augmentators/subject_object_augmentator.py

```diff
@@ -358,33 +439,66 @@ def is_eligible_for_augmentation(hun_graph: DependencyGraphWrapper, eng_graph: D
         Tests if a sentence (graph) pair is eligible for augmentation
         """
 
+        # Should contain one nsubj and one obj in both languages
+        if not SubjectObjectAugmentator.is_eligible_for_separate_augmentation(hun_graph, eng_graph, 'obj') or \
+                not SubjectObjectAugmentator.is_eligible_for_separate_augmentation(hun_graph, eng_graph, 'nsubj'):
+            return False
+
         hun_nsubj_edges = hun_graph.get_edges_with_property('dep', 'nsubj')
         eng_nsubj_edges = eng_graph.get_edges_with_property('dep', 'nsubj')
 
         hun_obj_edges = hun_graph.get_edges_with_property('dep', 'obj')
         eng_obj_edges = eng_graph.get_edges_with_property('dep', 'obj')
 
         # Should contain one nsubj and one obj in both languages
-        if len(hun_nsubj_edges) != 1 or len(eng_nsubj_edges) != 1 or len(hun_obj_edges) != 1 or len(eng_obj_edges) != 1:
-            return False
-        else:
-            hun_nsubj_edge = hun_nsubj_edges[0]
-            eng_nsubj_edge = eng_nsubj_edges[0]
-            hun_obj_edge = hun_obj_edges[0]
-            eng_obj_edge = eng_obj_edges[0]
+        #if len(hun_nsubj_edges) != 1 or len(eng_nsubj_edges) != 1 or len(hun_obj_edges) != 1 or len(eng_obj_edges) != 1:
+        #    return False
+        #else:
+        hun_nsubj_edge = hun_nsubj_edges[0]
+        eng_nsubj_edge = eng_nsubj_edges[0]
+        hun_obj_edge = hun_obj_edges[0]
+        eng_obj_edge = eng_obj_edges[0]
```

Comment:
Can we add a comment here to explain why we just take the first elements of each lists? (I assume it's because we constrained that there can be one dependency edge of that type only, but if we want to parametrize this in the future, this code will and will be easier to fix if we have some comments

Reply:
still needs fixing

Reply:
I added comments earlier